src/DDTrace/Span.php

```diff
@@ -134,6 +134,11 @@ public function setTag($key, $value, $setIfFinished = false)
         if ($key !== (string)$key) {
             throw InvalidSpanArgument::forTagKey($key);
         }
+        // Since sub classes can change the return value of a known method,
+        // we quietly ignore values that could cause errors when converting to string
+        if (is_object($value) && $key !== Tag::ERROR) {
```

Comment:
On second thought - would it make sense to check if object has `__toString` method ? and only ingore it if does not ?

Reply:
That's what [I was suggesting](https://github.com/DataDog/dd-trace-php/issues/332#issuecomment-487464713) too! :)

But @labbati made a good point about potentially encountering huge strings with, for example, giant ORM models and similar objects. If we really wanted to, we could do a check for string size and truncate it if it's too big. But I'm also thinking of a scenario where the tracer encounters 1,000+ huge models that get converted to a string for the trace. Then we have a ton of string juggling & memory consumption gong on. And since none of our officially supported integrations return objects that we want to use as tags (with the exception of `Exception`'s), we'd be committing to support a feature which might make it even harder to debug segfaults in very specific edge cases.

What are your thoughts? :)

Reply:
Yeah that makes a lot of sense. Thanks for clarifying